policies/push-global.rego

```diff
@@ -1,75 +1,80 @@
+propose {
+    project_affected
+}
+
+# Track if project files are affected
 track {
-    config_affected
+    project_affected
     input.push.branch == input.stack.branch
 }
 
-# Only trigger a stack run if the Terraform files were modified
-notrigger {
-  config_affected
-  not tf_affected
+# Track if stack config files are affected
+track {
+    stack_config_affected
+    input.push.branch == input.stack.branch
 }
 
-propose { tf_affected }
-ignore  {
-    not tf_affected
-    not config_affected
-}
-ignore  { input.push.tag != "" }
+# Get all of the affected files
+affected_files := input.push.affected_files
 
-# If pre-commit hooks make changes, they are not semantic changes
-# and can and should be ignored.
-ignore  { input.push.message == "pre-commit fixes" }
+# Track these extensions in the project folder
+tracked_extensions := {".tf", ".tf.json", ".tfvars", ".yaml"}
 
-# Fetch all of our affected files
-filepath := input.push.affected_files
+project_root := input.stack.project_root
 
-# Check if any Terraform files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".tf")
+# Check if any of the tracked extensions were modified in the project folder
+# https://www.openpolicyagent.org/docs/latest/policy-language/#some-keyword
+# https://www.openpolicyagent.org/docs/latest/policy-language/#variable-keys
+# https://www.openpolicyagent.org/docs/latest/policy-reference/#iteration
+project_affected {
+    some i, j
+    startswith(affected_files[i], project_root)
+    endswith(affected_files[i], tracked_extensions[j])
 }
 
-# Check if any Terraform json files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".tf.json")
-}
+# Split the stack name into a list
+stack_name_parts := split(input.stack.name, "-")
 
-# Check if any .tfvars files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".tfvars")
+# Check if the environment has been modified
+stack_config_affected {
+    contains(affected_files[_], concat("-", [stack_name_parts[0], stack_name_parts[1]]))
 }
 
-# Check if any .yaml files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".yaml")
-}
+# Get labels
+labels := input.stack.labels
 
-# Split our stack name into a list for matching below
-stack_name := split(input.stack.name, "-")
+# Get imports from the provided `labels`
+# https://www.openpolicyagent.org/docs/latest/policy-language/#comprehensions
+imports := [imp | startswith(labels[i], "import:"); imp := split(labels[i], ":")[1]]
 
-# Check if our global settings have been modified
-config_affected {
-    contains(filepath[_], "/globals.yaml")
+# Check if any of the imports have been modified
+stack_config_affected {
+    endswith(affected_files[_], imports[_])
```

Comment:
I am a little worried about using `endswith` here. What if the import is `eks-defaults` and the affected file is `old-eks-defaults`? `affected_files` should have a `/` delimiter, right? So we can use 
```suggestion
stack_config_affected {
    endswith(affected_files[_], sprintf("/%v", [imports[_]]))
```

Reply:
```
{
    "push": {
        "affected_files": [
            "stacks/uw2-globals.yaml",
            "stacks/uw2-devplatform.yaml",
            "components/terraform/vpc/main.tf"
        ],
        "branch": "main"
    },
    "stack": {
        "branch": "main",
        "labels": [
            "import:stacks/globals.yaml",
            "import:stacks/uw2-globals.yaml",
            "folder:uw2",
            "folder:uw2/devplatform",
            "folder:component/vpc"
        ],
        "name": "uw2-devplatform-vpc",
        "project_root": "components/terraform/vpc"
    }
}
```

both `affected_files` and `imports` start with `stacks` so it should be ok.

Reply:
```
  imports           = [for import in try(module.yaml_stack_config[each.key].config.0.imports, []) : format("%s/%s.yaml", local.stack_config_path, import)]
```